### ISSUE TITLE
Profile written with public key of other owned name

### ISSUE THREAD

**Issue by kantai** [blockstack/blockstack-browser#684]
Jay reported that a name he registered with the browser couldn't be looked up correctly. The name in question is `jay_cross.id` -- looking into it, the zonefile points to a profile in his dropbox, but that profile reports a public key which is not associated with the owner address.

You can see from these commands:

```
$ blockstack whois jay_cross.id
{
    ...
    "owner_address": "13hJJBuviynJirQ8Y9vXzJUpUzXvM7h4r5", 
    ...
}
```

The zonefile:
```
$ blockstack get_name_zonefile jay_cross.id
$ORIGIN jay_cross.id
$TTL 3600
_http._tcp URI 10 1 "https://www.dropbox.com/s/ybokh5eceme09ja/profile.json?dl=1"
```

And then that profile url:
```
[
  {
    "token": "...",
    "decodedToken": {
        ....
        "subject": {
          "publicKey": "02221b0f38b5cc5c748b18689efb20dc37798e5171799fbdcf9e45c59e305803fd"
        },
        "issuer": {
          "publicKey": "02221b0f38b5cc5c748b18689efb20dc37798e5171799fbdcf9e45c59e305803fd"
        },
        "claim": {
              ....
        }
     }
   }
 }
]
```

The public key corresponds to this address:
```
$ python -c "import keylib as k; print k.ECPublicKey('02221b0f38b5cc5c748b18689efb20dc37798e5171799fbdcf9e45c59e305803fd').address()"
163SeVgQt7rvHLMJuR4LhA7AULihKw5F28
```

This owns the name "hello_world.id"
```
$ blockstack get_names_owned_by_address 163SeVgQt7rvHLMJuR4LhA7AULihKw5F28
[
    "hello_world.id"
]
```

The profile reports to be using a public key associated with the owner of "hello_world.id", and the signature actually matches the key, which probably means that the profile is being signed by the key for one of the other names in the same browser instance.

@jay-hwang is `hello_world.id` the first name you registered in that browser?

**Comment by jay-hwang** [comment]
No, I registered a couple usernames before `hello_world.id`.
After I bought `jay_cross.id`, I restored `ithiel.id` and then `hello_world.id` on the same account. Maybe purchasing and restoring accounts has something to do with it?

**Comment by larrysalibra** [comment]
Thanks for the detailed bug report issue @kantai & @jay-hwang. This is really important to check out before the next release.

**Comment by larrysalibra** [comment]
Found the problem: profile updates are hardcoded to use the first key. This is some legacy code that didn't get changed when we moved to multiple profiles:
https://github.com/blockstack/blockstack-browser/blob/a1953ac29d6029bb6b29f9064a90ce87abfdbc9b/app/js/profiles/EditProfilePage.js#L94

I'll get a fix for this into the v0.14 build and it will be a matter of saving your profile again to fix it.

**Comment by larrysalibra** [comment]
@jay-hwang I pushed a fix for this. Once the new version comes out, you should be able to fix your profile by going to the edit page and clicking the save button which will sign it with the proper key and upload the new profile. Thank you again for reporting!
